Framework/MDAlgorithms/test/Integrate3DEventsTest.h

```diff
@@ -122,6 +123,260 @@ class Integrate3DEventsTest : public CxxTest::TestSuite {
       TS_ASSERT_DELTA(sigi, sigi_some[i], 0.01);
     }
   }
+
+  void test_integrateWeakPeakInPerfectCase() {
+    /* Check that we can integrate a weak peak using a strong peak in the
+     * perfect case when there is absolutely no background
+     */
+
```

Comment:
the generation could be replaced by a function call as it is repeated 3 times. You could just add some bools to determine number of peaks to add (2 or 3) and if to add a background